src/modules/35-connectors/components/CreateConnector/PdcConnector/StepDetails/PdcDetails.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useState, useEffect, useMemo } from 'react'
+import {
+  Layout,
+  Button,
+  Formik,
+  Text,
+  StepProps,
+  Container,
+  ButtonVariation,
+  HarnessDocTooltip,
+  FormInput
+} from '@wings-software/uicore'
+import { FontVariation } from '@harness/design-system'
+import type { ConnectorConfigDTO, ConnectorInfoDTO } from 'services/cd-ng'
+import { useStrings } from 'framework/strings'
+import UploadJSON from '../components/UploadJSON'
+
+import css from '../CreatePdcConnector.module.scss'
+
+interface PdcDetailsProps {
+  name: string
+  isEditMode: boolean
+  setIsEditMode: (val: boolean) => void
+  setFormData?: (formData: ConnectorConfigDTO) => void
+  onConnectorCreated: (data?: ConnectorConfigDTO) => void | Promise<void>
+  connectorInfo: ConnectorInfoDTO | void
+  accountId: string
+  orgIdentifier: string
+  projectIdentifier: string
+}
+
+export interface uploadHostItem {
+  hostname: string
+}
+interface StepConfigureProps {
+  closeModal?: () => void
+  onSuccess?: () => void
+  hosts?: string | []
+  spec?: { hosts: string }
+}
+
+const PdcDetails: React.FC<StepProps<StepConfigureProps> & PdcDetailsProps> = props => {
+  const { prevStepData, nextStep, isEditMode } = props
+  const { getString } = useStrings()
+
+  const [hostsJSON, setHostsJSON] = useState([] as uploadHostItem[])
+
+  useEffect(() => {
+    if (typeof prevStepData?.spec?.hosts === 'object' && !isEditMode) {
+      setHostsJSON(prevStepData?.spec?.hosts)
+    }
+  }, [])
+
+  const handleSubmit = (formData: ConnectorConfigDTO) => {
+    const data = { ...formData }
+    if (hostsJSON?.length > 0) {
+      data.hosts = hostsJSON
+    }
+    nextStep?.({ ...prevStepData, ...data } as StepConfigureProps)
+  }
+
+  const initialFormValues = useMemo(() => {
+    const hosts = prevStepData?.hosts || prevStepData?.spec?.hosts
+    return {
+      hosts: typeof hosts === 'string' ? hosts : hosts?.map?.((host: any) => host.hostname).join('\n')
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
      hosts: typeof hosts === 'string' ? hosts : hosts?.map?.((host: unknown) => host.hostname).join('\n')
```